src/cfnlint/data/AdditionalSpecs/RequiredBasedOnValue.json

```
@@ -0,0 +1,22 @@
+{
+    "PropertyTypes": {},
+    "ResourceTypes": {
+        "AWS::Lambda::EventSourceMapping": {
+            "EventSourceArn": [
+                {
+                    "Regex": "arn:aws:(kinesis|kafka|dynamodb):*",
```

[missing support for other partitions](https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html#arns-syntax)
```suggestion
                    "Regex": "arn:aws*:(kinesis|kafka|dynamodb):*",
```

---

@kddejong sorry, got confused by that trailing wildcard. Should be something like this instead:

```diff
+                    "Regex": "arn:aws.*:(kinesis|kafka|dynamodb):",
```

---

Oops.  Resolved it here #1759